metriq-api/controller/platformController.js

```diff
@@ -0,0 +1,63 @@
+// platformController.js
+
+// Service classes
+const PlatformService = require('../service/platformService')
+// Service instances
+const platformService = new PlatformService()
+
+function sendResponse (res, code, m) {
+  const body = JSON.stringify({ message: m })
+  res.writeHead(code, {
+    'Content-Length': Buffer.byteLength(body),
+    'Content-Type': 'text/plain'
+  })
+    .end(body)
+}
+
+async function routeWrapper (res, serviceFn, successMessage) {
+  try {
+    // Call the service function, to perform the intended action.
+    const result = await serviceFn()
+    if (result.success) {
+      // If successful, pass the service function result as the API response.
+      res.json({ message: successMessage, data: result.body }).end()
+    } else {
+      // The service function handled an error, but we can't perform the intended action.
+      sendResponse(res, 400, result.error)
+    }
+  } catch (err) {
+    // There was an unhandled exception.
+    sendResponse(res, 500, err)
+  }
+}
+
+// Validate the submission request and create the submission model.
```

Comment:
Should this comment pertain to "platform" as opposed to "submission"?

Reply:
Yes. (Done.)